### PR TITLE
docs(PeoplePicker): update PeoplePickerModal.md with new search functionality

### PR DESCRIPTION
**Pull request by dryhten**
`PeoplePickerModal.md` is outdated. The Search Box functionality has changed and should be reflected in the docs.

### CODE REVIEW

File: app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerModal.md [hunk 1]
```diff
@@ -4,20 +4,14 @@ A people picker in a modal
...
-Selecting a match from the dropdown populates the search box and updates the people pods returned.
-
-Alternatively, the user can elect to search for any full or partial name already entered without choosing from the dropdown list. This returns all matching person pods.
-
-The dropdown list is keyboard-navigable.
+To update the list of peron pods the user has to press enter or click on the search icon after typing something.
 
 Searching for an empty string returns all person pods.
 
+Currently the input won't generate a dropdown list of suggestions (but will at some point in the future).
```
Comment (diversemix) [inline app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerModal.md]: @dryhten When we tested this there were suggestions of name matches
@hdrury1 ... not sure there is a plan to do this in the future at the moment... I expect we will do if if someone asks?

Comment (dryhten) [reply app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerModal.md]: @diversemix what do you mean? After https://github.com/elifesciences/elife-xpub/pull/763 there is no list of suggestions. Also, check https://github.com/elifesciences/elife-xpub/issues/629#issuecomment-428088027

Comment (diversemix) [reply app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerModal.md]: Yes you are right just tested locally... sorry ..  probably confused from when I was testing before !

Comment (hdrury1) [reply app/components/pages/SubmissionWizard/steps/Editors/PeoplePickerModal.md]: Original designs included an auto suggest drop down. But until we start getting real world feedback I remain unconvinced that we actually need one (Journal doesn’t currently have one). In any case the product recs are far from finalised so I’d rather remove what we have now (which isn’t an auto suggest anyway).

**Pull request merged**